tests/unit/spec/base/send-to-device.js

```diff
@@ -184,70 +228,132 @@ describe('send-to-device.js', function() {
     describe('onFormSubmit', function() {
 
         beforeEach(function() {
-            spyOn($, 'get').and.callFake(function () {
-                var d = $.Deferred();
+            this.firstGetCall = function(){
                 var data = {
                     country_code: 'us'
                 };
-                d.resolve(data);
-                return d.promise();
-            });
+                var pd = window.Promise.resolve(
+                    new Response(
+                        new Blob([JSON.stringify(data)])
+                    )
+                );
+                return pd;
+            };
+
+            // Specifies the delay to use in setTimeout for `expect` calls
```

Comment:
This feels like something that a promise might resolve, versus using a setTimeout?

Reply:
Using a fetch().then().then() inside the top level function necessitated the use of a setTimeout as the top level function doesn't directly return a promise.

I tried other approaches with done, and async but none of them worked as intended.

However, now that I can't use fetch, these tests will again change